ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -1099,6 +1099,86 @@ def test_ipahealthcheck_ds_riplugincheck(self, update_riplugin):
             assert check["result"] == "WARNING"
             assert warn_msg in check["kw"]["msg"]
 
+    @pytest.fixture
+    def update_logging(self):
+        """
+        Fixture disables nsslapd-logging-hr-timestamps-enabled
+        parameter and reverts it back
+        """
+        ldap = self.master.ldap_connect()
+        dn = DN(
+            ("cn", "config"),
+        )
+        entry = ldap.get_entry(dn)  # pylint: disable=no-member
+        entry.single_value["nsslapd-logging-hr-timestamps-enabled"] = 'off'
+        ldap.update_entry(entry)  # pylint: disable=no-member
+
+        yield
+
+        entry = ldap.get_entry(dn)  # pylint: disable=no-member
+        entry.single_value["nsslapd-logging-hr-timestamps-enabled"] = 'on'
+        ldap.update_entry(entry)  # pylint: disable=no-member
+
+    def test_ipahealthcheck_ds_configcheck(self, update_logging):
+        """
+        This testcase ensures that ConfigCheck displays warning
+        when high resolution timestamp is disabled.
+        """
+        warn_msg = (
+            "nsslapd-logging-hr-timestamps-enabled changes the "
+            "log format in directory server "
+        )
+        returncode, data = run_healthcheck(
+            self.master,
+            "ipahealthcheck.ds.config",
+            "ConfigCheck",
+        )
+        assert returncode == 1
+        for check in data:
+            assert check["result"] == "WARNING"
+            assert check["kw"]["key"] == "DSCLE0001"
+            assert 'cn=config' in check["kw"]["items"]
+            assert warn_msg in check["kw"]["msg"]
+
+    @pytest.fixture
+    def permission_change(self):
+        """
+        Fixture to modify the permission of pwdfile.txt to 000
+        revert back
+        """
+        instance = realm_to_serverid(self.master.domain.realm)
+        cmd = [
+            "chmod",
+            "000",
+            paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance + "/pin.txt"]
+        self.master.run_command(cmd)
+        yield
+        cmd = [
+            "chmod",
+            "400",
+            paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance + "/pin.txt",
+        ]
+        self.master.run_command(cmd)
```

Comment:
You can re-use the existing modify_permissions fixture. See ```test_ipa_filecheck_bad_too_restrictive``` as a usage example.

Reply:
moved the test_ipahealthcheck_ds_fschecks under TestIpaHealthCheckFileCheck and also used the existing modify_permisson fixture for the test.